base/datafmt.jl

```diff
@@ -463,15 +463,15 @@ function colval{T<:Char}(sbuff::String, startpos::Int, endpos::Int, cells::Array
 end
 colval(sbuff::String, startpos::Int, endpos::Int, cells::Array, row::Int, col::Int) = true
 
-function dlm_parse{T,D}(dbuff::T, eol::D, dlm::D, qchar::D, cchar::D,
-                        ign_adj_dlm::Bool, allow_quote::Bool, allow_comments::Bool,
-                        skipstart::Int, skipblanks::Bool, dh::DLMHandler)
+function dlm_parse{D}(dbuff::String, eol::D, dlm::D, qchar::D, cchar::D,
+                      ign_adj_dlm::Bool, allow_quote::Bool, allow_comments::Bool,
+                      skipstart::Int, skipblanks::Bool, dh::DLMHandler)
     all_ascii = (D <: UInt8) || (isascii(eol) &&
                                  isascii(dlm) &&
                                  (!allow_quote || isascii(qchar)) &&
                                  (!allow_comments || isascii(cchar)))
-    if T === String && all_ascii
-        return dlm_parse(dbuff.data, eol % UInt8, dlm % UInt8, qchar % UInt8, cchar % UInt8,
+    if all_ascii && !(D <: UInt8)
```

Comment:
I don't know. The whole `all_ascii` business at the top of this function can probably be removed.